cclib/parser/data.py

```diff
@@ -68,6 +68,7 @@ class ccData:
         nbasis -- number of basis functions (integer)
         nmo -- number of molecular orbitals (integer)
         nmrtensors -- Nuclear magnetic resonance chemical shielding tensors (dict of dicts of array[2])
+        nmrcouplingtensors -- Nuclear magnetic resonance spin-spin coupling tensors (dict of dicts of array[2])

```

Comment:
Should these both not be `dict of dicts of array[3]`?

Reply:
The number is the number of array axes/the tensor rank; how many indices are required to access an element?  1 for a vector, 2 a matrix, 3 a cube, etc.

Reply:
Aye I got you, think this has confused me before for some reason